dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -357,15 +359,30 @@ class AsyncRequestHandler : public UnaryCallback<bool>
             setDone(done_msg);
     }
 
-    bool sendPackets()
+    bool sendPackets(std::string & err_info)
     {
         for (size_t i = 0; i < read_packet_index; ++i)
         {
             auto & packet = packets[i];
-            if (!pushPacket<enable_fine_grained_shuffle, false>(request->source_index, req_info, packet, *msg_channels, log))
+            // We shouldn't throw error directly, since the caller works in a standalone thread.
+            try
+            {
+                packet->recomputeTrackedMem();
```

Comment:
Why not recompute tracked mem in `execute`, since if we recompute in here, we will get the memory exceeded error after a batch packets are read, and if we recompute in `execute`, we will get the memory exceeded error after one packet is read.

Reply:
wait another PR to solve error-handle problem...